crates/sui-core/src/authority_aggregator.rs

```diff
@@ -1213,6 +1231,16 @@ where
             };
         }
 
+        if state.tx_signatures.total_votes() + state.retryable_overloaded_stake >= quorum_threshold
```

Comment:
i don't think this is right - it doesn't prove that additional votes cannot arrive such that total_votes() exceeds quorum_threshold.

Reply:
It's very subtle. When the code arrives here, we have heard from *all* validator's response...

The code block here [https://github.com/MystenLabs/sui/blob/54c06347f9ae3f118c85cefe6bdf6e65676f865b/crates/sui-core/src/authority_aggregator.rs#L1040-L1115](url) handles errors from validator. Any early exit of error handling will set `state.retryable` to false. The fact that we arrived here with `state.retryable == true` indicates that either total_votes() is >= 2f+1, or we have heard from *all* validators.

Reply:
Comments added.